modules/channel-web/src/backend/api.ts

```diff
@@ -161,13 +161,20 @@ export default async (bp: typeof sdk, db: Database) => {
   )
 
   router.post(
-    '/messages',
+    '/messages/:socketId',
     bp.http.extractExternalToken,
     assertUserInfo(),
     asyncMiddleware(async (req: ChatRequest, res: Response) => {
       const { botId, userId } = req
       let { conversationId } = req
 
+      const socketId = req.params.socketId
+      const visitorId = await bp.realtime.getVisitorIdFromSocketId(socketId)
+
+      if (!visitorId || userId !== visitorId) {
+        return res.sendStatus(403)
+      }
+
```

Comment:
I suggest adding socketId to the body and implement the validation un "assertUserInfo". This way, we have a single method which validates every request